### PR TITLE
Update idris.tmLanguage with 0.99 syntax

### PR DESCRIPTION
**Pull request by be5invis**

### CODE REVIEW

File: syntaxes/idris.tmLanguage [hunk 1]
```diff
@@ -121,6 +121,12 @@
 			</array>
 		</dict>
 		<dict>
+			<key>match</key>
+			<string>(?&lt;!\w)\'\w++(?!')</string>
```
Comment (zjhmale) [inline syntaxes/idris.tmLanguage]: What exactly the use case of the mandatory green match?

Comment (be5invis) [inline syntaxes/idris.tmLanguage]: otherwise it will match `'abcde` in text `'abcdef'`.

**Comment by zjhmale** [comment]
Hi @be5invis, sorry for the delay, I was busy these days and did not check out GitHub notifications recently 😅

And for the syntax file, I planned to sync this from [atom-package](https://github.com/idris-hackers/atom-language-idris/blob/master/grammars/language-idris.cson), need to find a way to translate from CSON file to JSON file safely. Because that plugin will be catch up the changes of Idris itself and maintaining another tmLanguage file by our own seems redundant here.

I will merge this first, and then we can discuss the proposal above later 😄

File: syntaxes/idris.tmLanguage (hunk 1 above)
Comment (zjhmale) [inline syntaxes/idris.tmLanguage]: Ok, make sense.

Comment (zjhmale) [inline syntaxes/idris.tmLanguage]: Sorry, I don't get the point, I think currently `'abcdef'` just works like a charm

image

**Comment by be5invis** [comment]
@zjhmale Hmm maybe I should add support for `rewrite <term> in <term>` either.

**Comment by be5invis** [comment]
@zjhmale and `tmLanguage` is a PLIST file, so that you can convert CSON to JSON first, and then turn it into PLIST.

**Comment by zjhmale** [comment]
Thanks for the tip! Actually, VSCode can also work with JSON-defined syntax as expected, so maybe just convert from CSON to JSON is kindly enough 😄 , what do you think?
>@zjhmale and tmLanguage is a PLIST file, so that you can convert CSON to JSON first, and then turn it into PLIST.

File: syntaxes/idris.tmLanguage (hunk 1 above)
Comment (zjhmale) [inline syntaxes/idris.tmLanguage]: Can you give me some examples of this syntax change? I just can not get my sense to embrace this change actually 😅.

Comment (be5invis) [inline syntaxes/idris.tmLanguage]: Idris have introduced tagged effects using this syntax:
```idris
whatever : Eff ['a ::: STATE, 'b ::: STATE] ()
whatever = do
    x <- ('a :- get)
    'b :- (put x)
```
So we have to support `'identifier` syntax.

Comment (zjhmale) [inline syntaxes/idris.tmLanguage]: Cool, just not catch up with the community for months, thanks @be5invis.

**Pull request merged**